axum/src/response/sse.rs

```diff
@@ -407,6 +406,48 @@ impl Event {
     }
 }
 
+impl std::io::Write for EventDataWrite<'_> {
```

Comment:
Why are we implementing `std::io::Write` instead of `std::fmt::Write`? Given that event streams are required to be valid UTF-8, I think it makes more sense to implement the latter.

For the JSON case, we could maybe avoid UTF-8 checks with a wrapper, since we know the concatenated output will be valid UTF-8?